gpMgmt/bin/gppylib/logfilter.py

```diff
@@ -543,29 +553,31 @@ def TimestampInBounds(iterable, begin, end):
 
     # Fetch first item from input stream.
     source = iter(iterable)
-    item = next(source)
-
-    # If first item is a string, assume input consists of individual lines.
-    # Yield lines which start with a timestamp within the given bounds, plus
-    # any following lines which do not have timestamps.
-    if isinstance(item, str):
-        withinbounds = False
-        for item in source:
+    try:
+        first_item = next(source)
+    except StopIteration:
+        return
+
+    withinbounds = False
+    for item in chain(iter([first_item]), source):
```

Comment:
why not simply
```suggestion
    for item in source:
```
without above `next(source)`